_rules/image-not-in-acc-tree-is-decorative-e88epe.md

```diff
@@ -0,0 +1,300 @@
+name: Image not in the accessibility tree is decorative
+rule_type: atomic
+description: |
+  This rule checks that visible img, svg and canvas elements that are ignored by assistive technologies are decorative
+accessibility_requirements:
+  wcag20:1.1.1: # Non-text content (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - Accessibiliy tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Brian Bors
+    - Wilco Fiers
+  assets:
+    - W3C
+    - Wikimedia
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # (used with `npm test` to ensure validity of code snippets)
+  - 'attr-lowercase'
+---
+
+## Applicability
+
+Any `img`, `canvas` or `svg` element that is [visible][] and is not [included in the accessibility tree][], except if one of the following is true:
+
+- **unloaded img**: the element is an `img` and its [current request][]'s [state][image request state] is not [completely available][]; or
+- **ignored svg**: the element is an `svg` with an empty (`""`) [accessible name][] and a [semantic role][] of `graphics-document`; or
+- **ignored canvas**: the element is a `canvas` with an empty (`""`) [accessible name][] and no [explicit semantic role][]; or
+- **named from author**: the element is a [descendant][] in the [flat tree][] of an element that is [named from author][].
+
+**Note**: An example of an image ignored because of "named from author" is when the image is a descendant of a `button` element that uses `aria-label` for its accessible name.
+
+## Expectation
+
+Each test target is [purely decorative][].
+
+**Note**: It is relatively common for an informative image such as an icon to be marked up as decorative, if the text alternative is adjacent to the image. This is a [conforming alternative version][] for the image. This fails the rule but meets [conformance requirement 1 of WCAG 2.1](https://www.w3.org/TR/WCAG21/#cc1).
+
+## Assumptions
+
+- `svg` elements with a [semantic role][] of `graphics-document` and with an empty (`""`) [accessible name][] are ignored by assistive technologies tested for this rule. If some assistive technology does not ignore these elements, and that assistive technology is required for conformance, passing this rule does not ensure all decorative `svg` elements can be ignored, and the [success criterion 1.1.1 Non-text content][] may still not be satisfied. The same is true for `canvas` elements with no [semantic role][] and an empty (`""`) [accessible name][].
+
+- A web page with informative images without an [accessible name][] may conform to WCAG 2.1 Level A when the information provided by that image is available elsewhere, either on the web page itself or on a [conforming alternative version][] of the page. For example if an equivalent text is adjacent to the image, or if the text alternative is included in the [accessible name][] of a parent element.
```

Comment:
As much as I agree we need the "no conforming alternative version of the image" part, I am not sure we need the "no conforming alternative version of the *page*" part.

Ain't all rules assuming that the test subject is what need to be tested in the first place and that the is no conforming alternative version of it?

Reply:
You are correct, rules generally do not take confirming alternative versions into account. I talked with Brian about this for quite some time. The issue is something like this:

```html
<img src=".." alt="pdf" />Hello world PDF
```

Some folks call this out as a failure. Their argument is that the image is redundant, and thus decorative and fails because it won't be ignored by assistive technologies. That's not the stance we're taking for this rule. Here we're saying that even if the information is redundant, the image is still informative.  If you change the alt to an empty alt, a conforming alternative version is needed. Problem there is CAV is rather vague, and depending on how you read it, might only be applicable to different pages. We're making an assumption in this rule that this isn't the case, that a CAV can be in the same page, and that this interpretation on redundant information not being decorative is correct.

Reply:
Yes, this is an example of a case of "conforming alternative version on the same page" case which I agree is needed.

What I'm saying is that the "no conforming alternative version **of the full page**" part is not needed.
Essentially, if we have
```html
page1.html:
<img src="important-image.jpg" alt="" />

page1-conforming-alternative.html:
<img src="important-image.jpg" alt="A descriptive alt text" />
```
Then the rule is going to fail on `page1.html`, but the full thngie is good because there is a conforming alternative version of it, so the failure of the rule is due to the assumption "no conforming alternative version of the page" being false.

But, this is essentially the case for every single rule. If the rule is tested on a page which has a conforming alternative version of it, then yeah, the rule is going to fail.

So far, I considered that
> A web page [failing this rule] may conform to WCAG 2.1 when the information is available elsewhere, on a [conforming alternative version][] of the page.

was essentially a global assumption made by every single rule and doesn't need to be repeated in each of them.
(e.g. "A web page with no `title` may conform to WCAG if there is a conforming alternative version of the page with a `title`", and there is actually no need to put that assumption in the title rules)

That's why I'm suggesting restricting this assumption to
```suggestion
- `svg` elements with a [semantic role][] of `graphics-document` and with an empty (`""`) [accessible name][] are ignored by assistive technologies tested for this rule. If some assistive technology does not ignore these elements, and that assistive technology is required for conformance, passing this rule does not ensure all decorative `svg` elements can be ignored, and the [success criterion 1.1.1 Non-text content][] may still not be satisfied. The same is true for `canvas` elements with no [semantic role][] and an empty (`""`) [accessible name][].

- A web page with informative images without an [accessible name][] may conform to WCAG 2.1 Level A when the information provided by that image is available elsewhere on the web page itself. For example if an equivalent text is adjacent to the image, or if the text alternative is included in the [accessible name][] of a parent element.
```